meson.build

```diff
@@ -41,7 +41,7 @@ project('picolibc', 'c',
           'warning_level=2',
 	],
 	license : 'BSD',
-	meson_version : '>= 0.53',
+	meson_version : '>= 0.57',
```

Comment:
maybe we can make this transition now -- we kept this version to support common Ubuntu environments. I'll go re-review the decisions from several years ago about this.